dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/model/ProteinFeature.java

```diff
@@ -15,41 +15,33 @@
  * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.icgc.dcc.imports.gene.util;
-
-import java.util.Set;
-
-import org.icgc.dcc.imports.gene.core.GeneFilter;
+package org.icgc.dcc.imports.gene.model;
 
+import lombok.Data;
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
-import lombok.val;
-
-import com.fasterxml.jackson.databind.JsonNode;
-
-@RequiredArgsConstructor
-public class InGeneFilter implements GeneFilter {
 
-  /**
-   * Configuration.
-   */
-  @NonNull
-  private final Set<String> geneIds;
-
-  @Override
-  public boolean filter(JsonNode gene) {
-    val text = gene.toString();
-    for (val geneId : geneIds) {
-      if (text.contains(geneId)) {
-        return true;
-      }
-    }
-
-    return false;
+/**
+ * POJO Class for Protein Features
+ */
+@Data
+public class ProteinFeature {
+
+  private String interproId;
+  private String hitName;
+  private String gffSource;
+  private String description;
+  private String analysisId;
+  private int start;
+  private int end;
+
+  public ProteinFeature(@NonNull String interproId, @NonNull String hitName, @NonNull String description) {
+    this.interproId = interproId;
+    this.hitName = hitName;
+    this.description = description;
   }
 
-  public static InGeneFilter in(Set<String> geneIds) {
-    return new InGeneFilter(geneIds);
+  public ProteinFeature copy() {
```

Comment:
Checkout `@Wither`. Works nice with `@Value` to make "streamable" immutable objects.